mobile/lib/ui/viewer/search/result/people_section_all_page.dart

```diff
@@ -55,67 +101,76 @@ class _PeopleSectionAllPageState extends State<PeopleSectionAllPage> {
-            return Center(child: Text(S.of(context).noResultsFound + '.'));
-          } else {
-            final results = snapshot.data!;
-            final screenWidth = MediaQuery.of(context).size.width;
-            final crossAxisCount = (screenWidth / 100).floor();
-
-            final itemSize = (screenWidth -
-                    ((horizontalEdgePadding * 2) +
-                        ((crossAxisCount - 1) * gridPadding))) /
-                crossAxisCount;
-
-            return GridView.builder(
-              padding: const EdgeInsets.fromLTRB(
-                horizontalEdgePadding,
-                16,
-                horizontalEdgePadding,
-                96,
-              ),
-              gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
-                mainAxisSpacing: gridPadding,
-                crossAxisSpacing: gridPadding,
-                crossAxisCount: crossAxisCount,
-                childAspectRatio:
-                    itemSize / (itemSize + (24 * textScaleFactor)),
-              ),
-              itemCount: results.length,
-              itemBuilder: (context, index) {
-                return PersonSearchExample(
-                  searchResult: results[index],
-                  size: itemSize,
-                )
-                    .animate(delay: Duration(milliseconds: index * 13))
-                    .fadeIn(
-                      duration: const Duration(milliseconds: 225),
-                      curve: Curves.easeIn,
-                    )
-                    .slide(
-                      begin: const Offset(0, -0.06),
-                      curve: Curves.easeInOut,
-                      duration: const Duration(
-                        milliseconds: 225,
-                      ),
-                    );
-              },
+
+    return FutureBuilder<List<GenericSearchResult>>(
+      future: sectionData,
+      builder: (context, snapshot) {
+        if (snapshot.connectionState == ConnectionState.waiting) {
+          return const Center(child: EnteLoadingWidget());
+        } else if (snapshot.hasError) {
+          return const Center(child: Icon(Icons.error_outline_rounded));
+        } else if (!snapshot.hasData || snapshot.data!.isEmpty) {
+          return Center(child: Text(S.of(context).noResultsFound + '.'));
+        } else {
+          final results = snapshot.data!;
+          final screenWidth = MediaQuery.of(context).size.width;
+          final crossAxisCount = (screenWidth / 100).floor();
+
+          final itemSize = (screenWidth -
+                  ((horizontalEdgePadding * 2) +
+                      ((crossAxisCount - 1) * gridPadding))) /
+              crossAxisCount;
+
+          List<GenericSearchResult> data = results;
+
+          if (widget.namedOnly && widget.selectedPeople != null) {
+            data = data.sorted(
+              (a, b) => widget.selectedPeople!.isPersonSelected(
+                (b.hierarchicalSearchFilter as FaceFilter).personId!,
```

Comment:
Get `personId` from `params` of `GenericSearchResult`

Reply:
Done!